src/exception/treatment-detail-exceptions/service.ts

```diff
@@ -74,3 +74,43 @@ export const create = async ({
 		return treatmentDetailExceptionResult;
 	}
 };
+
+export const deleteIdsByProgramId = async ({
+	programId,
+	donorSubmitterIds,
+	isDryRun,
+}: {
+	programId: string;
+	donorSubmitterIds: string[];
+	isDryRun: boolean;
+}): Promise<Result<DeleteResult>> => {
+	const treatmentDetailExceptionResult = await getByProgramId(programId);
+	if (treatmentDetailExceptionResult.success) {
+		const currentDonorIds = treatmentDetailExceptionResult.data.donorSubmitterIds;
+		const updatedDonorIds = currentDonorIds.filter((id) => !donorSubmitterIds.includes(id));
+
+		// calc deleted and unchanged ids
+		const donorsDeleted = donorSubmitterIds.filter((id) => currentDonorIds.includes(id));
+		const donorsUnchanged = currentDonorIds.filter((id) => !donorSubmitterIds.includes(id));
```

Comment:
further investigation this code looks like it mirrors entity exception pretty closely, and that code does the same thing.

Reply:
Ah right it's duplicated. Rather than dismiss your review, I will change this in the next PR for: https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/gh/icgc-argo/argo-clinical/1159